src/main/docs/guide/globalConfiguration.adoc

```diff
@@ -0,0 +1,32 @@
+The generation can be configured globally with annotation processor options:
+
+++++
+include::src/main/docs/guide/annotationProcessorOption.html[]
+++++
+
+<1> Set the base URL for all the schemas. It will be prepended to all relative schema URLs.
```

Comment:
It looks like this now:
<img width="1140" alt="image" src="https://github.com/micronaut-projects/micronaut-json-schema/assets/80816836/8916862c-76eb-4d41-bfca-50da2ff17705">
<img width="1124" alt="image" src="https://github.com/micronaut-projects/micronaut-json-schema/assets/80816836/40078b88-ceda-4466-b671-0948f1049afd">

Since the `baseUri` is an important option, it would seem strange to not mention how it can be specified. I probably spent like an hour finding out how annotation processing options are specified for different languages and build tools, so others are likely to have the same issue if we do not mention it.

@graemerocher Perhaps we should support a better way of configuring annotation processing? Like a `processing.properties` file that gets read from resources. It sounds strange, but at least it would provide a unique way of setting the options.

Reply:
this seems like a common challenge that both the openapi module and this module probably share, it is probably possible to make it simpler in Gradle not sure if it is possible in Maven

//cc @melix @alvarosanchez

Reply:
btw shouldn't the Maven config prefix be `-A` not `-`

Reply:
It is `-A`, yes.

What exactly is expected to be done with Maven?

Reply:
> What exactly is expected to be done with Maven?

Somehow load the annotation processor options from a properties file or somewhere else so that there would be a unified way to configure them for both gradle and maven.

Reply:
In Maven, it's the Maven Compiler Plugin the one that compiles, and this is how you pass annotation processing parameters: https://maven.apache.org/plugins/maven-compiler-plugin/examples/pass-compiler-arguments.html

Theoretically and hypothetically, we could create a Maven extension that, before compilation, would monkey patch the Maven Compiler Plugin configuration XML at runtime if a `processing.properties` file is found. It is a can of worms inside a box of snakes.

Reply:
I was thinking we can add options in here: https://github.com/micronaut-projects/micronaut-core/blob/a04877b58f5c4e98b91ffec498fa48b25c68785a/inject-java/src/main/java/io/micronaut/annotation/processing/visitor/JavaVisitorContext.java#L441-L449. 

It could probably be something like:
```java
FileObject fileObject = processingEnvironment.getFiler()
    .getResource( StandardLocation.CLASS_OUTPUT, "", "config.json" );
InputStream jsonStream = fileObject.openInputStream();
```
which relies on the fact that resources should be copied to the build folder. But I guess current GroovyVisitorContext won't be able to do this. And I am not sure if the behavior would be consistent for Maven & Groovy, so yeah, it's complicated.

Reply:
issue: I don't think we should do this. If a file shows up in the output classes folder, then it will end in the user application, where it has nothing to do. I think this should be solved on the build plugins side. There are several options for Gradle, which are not mutually exclusive:

1. add a generic option under the `processing` block of the `micronaut` configuration block to pass parameters
2. (preferred) use a strongly-typed DSL for json-schema, which would automatically populate the annotation processor parameters. This option has the benefit that if options of the processor are, say, input files, then we can make them part of inputs and behave correctly wrt up-to-date checking, which is not the case with stringly typed arguments.

Reply:
Thanks @melix I guess my main issue with this is that to update an option will require changes to 3 repositories: this repository, gradle plugin repository and maven plugin repository. And the documentation would also be in different places, too.

I agree that other solutions seem inferior, so I suppose this is what I will have to do.